src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1,16 +1,879 @@
+                self.cargo_door_timer -= *delta_time_update;
+            } else {
+                self.cargo_door_timer = Duration::from_secs(0);
+            }
+        }
+
+        self.cargo_door_timer > Duration::from_secs_f64(0.0)
+    }
+
+    pub fn is_cargo_operation_ptu_flag(&mut self, delta_time_update: &Duration) -> bool {
+        let cargo_door_moved = self.cargo_door_back_pos != self.cargo_door_back_pos_prev
+            || self.cargo_door_front_pos != self.cargo_door_front_pos_prev;
+
+        if cargo_door_moved {
+            self.cargo_door_timer_ptu =
+                Duration::from_secs_f64(A320HydraulicLogic::CARGO_OPERATED_TIMEOUT_PTU);
+        } else {
+            if self.cargo_door_timer_ptu > *delta_time_update {
+                self.cargo_door_timer_ptu -= *delta_time_update;
+            } else {
+                self.cargo_door_timer_ptu = Duration::from_secs(0);
+            }
+        }
+
+        self.cargo_door_timer_ptu > Duration::from_secs_f64(0.0)
+    }
+
+    pub fn is_nsw_pin_inserted_flag(&mut self, delta_time_update: &Duration) -> bool {
+        let pushback_in_progress =
+            (self.pushback_angle != self.pushback_angle_prev) && self.pushback_state != 3.0;
+
+        if pushback_in_progress {
+            self.nws_tow_engaged_timer =
+                Duration::from_secs_f64(A320HydraulicLogic::NWS_PIN_REMOVE_TIMEOUT);
+        } else {
+            if self.nws_tow_engaged_timer > *delta_time_update {
+                self.nws_tow_engaged_timer -= *delta_time_update; //TODO CHECK if rollover issue to expect if not limiting to 0
+            } else {
+                self.nws_tow_engaged_timer = Duration::from_secs(0);
+            }
+        }
+
+        self.nws_tow_engaged_timer > Duration::from_secs(0)
+    }
+}
+
+impl SimulationElement for A320HydraulicLogic {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        visitor.visit(self);
+    }
+
+    fn read(&mut self, state: &mut SimulatorReader) {
+        self.parking_brake_applied = state.read_bool("PARK_BRAKE_ON");
+        self.eng_1_master_on = state.read_bool("ENG MASTER 1");
+        self.eng_2_master_on = state.read_bool("ENG MASTER 2");
+        self.weight_on_wheels = state.read_bool("SIM ON GROUND");
+
+        //Handling here of the previous values of cargo doors
+        self.cargo_door_front_pos_prev = self.cargo_door_front_pos;
+        self.cargo_door_front_pos = state.read_f64("CARGO FRONT POS");
+        self.cargo_door_back_pos_prev = self.cargo_door_back_pos;
+        self.cargo_door_back_pos = state.read_f64("CARGO BACK POS");
+
+        //Handling here of the previous values of pushback angle. Angle keeps moving while towed. Feel free to find better hack
+        self.pushback_angle_prev = self.pushback_angle;
+        self.pushback_angle = state.read_f64("PUSHBACK ANGLE");
+        self.pushback_state = state.read_f64("PUSHBACK STATE");
```

Comment:
**suggestion**: Create a type into which you push the new value. The type should tell you if the value was changed compared to last time:

```rust
struct ObserveChangeVariable<T> {
    value: T,
    previous_value: T,
}
impl<T> ObserveChangeVariable<T> {
    fn has_changed(&self) {
        self.value != self.previous_value;
    }

    fn set_value(&mut self, value: T) {
        self.previous_value = self.value;
        self.value = value;
    }
}
```
Keeps things cleaner here 😃

Reply:
Discussed via Discord. For now we'll keep it as is.